mtags/src/main/scala-2/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -131,57 +150,113 @@ object MultilineStringFormattingProvider {
+        }(text, newlineAdded, tokens)
+      }
+      .getOrElse(Nil)
   }
 
-  private def shouldFormatMultiString(
-      sourceText: String,
-      tokens: Tokens,
-      start: StartPosition,
-      end: EndPosition
-  ): PartialFunction[(Token, Int), Boolean] = {
-    case (token: Constant.String, index: Int) =>
-      isMultilineString(sourceText, token) && hasStripMarginSuffix(
-        index,
-        tokens
-      ) && inToken(start, end, token)
+  private def isStringOrInterpolation(
+      startPosition: StartPosition,
+      endPosition: EndPosition,
+      andMultilineString: Boolean,
+      andHadStripMargin: Boolean,
+      andHasPipeInScope: Boolean
+  )(indent: => List[TextEdit])(
+      implicit sourceText: String,
+      newlineAdded: Boolean,
+      tokens: Tokens
+  ): PartialFunction[(Token, Int), List[TextEdit]] = {
+    case (token: Token.Constant.String, _)
+        if inToken(startPosition, endPosition, token)
+          && (if (andMultilineString) isMultilineString(token) else true)
+          && (if (andHadStripMargin) hasStripMarginSuffix(token) else true)
+          && (if (andHasPipeInScope)
+                pipeInScope(
+                  startPosition,
+                  endPosition,
+                  sourceText,
+                  newlineAdded
+                )
+              else true) =>
+      indent
+    case (token: Interpolation.Start, index: Int)
+        if token.pos.start < startPosition.start && {
+          var endIndex = index + 1
+          while (!tokens(endIndex)
+              .isInstanceOf[Interpolation.End]) endIndex += 1
+          val endToken = tokens(endIndex)
+          endToken.pos.end > endPosition.end &&
+          (if (andMultilineString) isMultilineString(token) else true) &&
+          (if (andHadStripMargin) hasStripMarginSuffix(endToken) else true) &&
+          (if (andHasPipeInScope)
+             pipeInScope(
+               startPosition,
+               endPosition,
+               sourceText,
+               newlineAdded
+             )
+           else true)
+        } =>
+      indent
   }
-  private def shouldFormatInterpolationString(
-      sourceText: String,
-      tokens: Tokens,
+
+  private def indentTokensOnTypeFormatting(
       start: StartPosition,
-      endPosition: EndPosition
-  ): PartialFunction[(Token, Int), Boolean] = {
-    case (token: Interpolation.Start, index: Int)
-        if token.pos.start <= start.start => {
-      var endIndex = index + 1
-      while (!tokens(endIndex)
-          .isInstanceOf[Interpolation.End]) endIndex += 1
-      isMultilineString(sourceText, token) && hasStripMarginSuffix(
-        endIndex,
-        tokens
-      ) && tokens(endIndex).pos.end > endPosition.end
-    }
+      end: EndPosition,
+      tokens: Tokens,
+      newlineAdded: Boolean,
+      splitLines: Array[String],
+      position: Position,
+      text: String
+  ): List[TextEdit] = {
+    implicit val sourceText: String = text
```

Comment:
Let's not use implicit here. They should never be used for simple types like strings or Boolean. It adds a level of complexity, you need to search for implicit rather than just have it there explicitly.  And this can also cause some unexpected bugs. Honestly, I think the simpler we can write the easier it will be for someone to read this in the future.

Let's just add explicit values. You can also just use currying or overloaded functions if you don't want too write those values multiple times.

```scala
val fun =  isStringOrInterpolation(
          start,
          end,
          andMultilineString = true,
          andHadStripMargin = true,
          andHasPipeInScope = true
        )(List(indent(splitLines, position))).orElse(...)

fun(sourceText, tokens, newlineAdded)
```

Reply:
I agree with you about implicit, but It makes the call so long. It was just a proposition.